src/ui/handler/box_zoom.js

```diff
@@ -147,13 +147,20 @@ class BoxZoomHandler {
         }
     }
 
-    _finish() {
-        this._active = false;
-
+    _unbind() {
```

Comment:
Let's guard against the race condition/double invocation that could occur when the user removes the map while a handler is active.

Reply:
I don't think thats possible, infact this PR would that from happening.
Scenario one:
If a handler callback is executing, `map.remove()` can't be called, because well, JS is single threaded.
Scenario two:
And if `map.remove()` is called while a handler is still active i.e the user has the mouse held down,  the callback gets unbound, so it would never invoke once the user eventually say, does a mouseup. Previous to this it would actually invoke the mouseup.

Reply:
> If a handler callback is executing, `map.remove()` can't be called, because well, JS is single threaded.
You can call `map.remove()` from the callback ;)

Reply:
Yea.. but why would we ever do that?

Reply:
It might not be intentional, e.g. an application could transition away from a map view (and calling `map.remove()`) while the user is still interacting with it.

Reply:
But that would be Scenario 2, which this fixes? `map.remove()` would still not be synchronously called in our callback, because thats our code and not user code?
That transition away would run in a different run-loop tick, so any of the enteraction callbacks would have been fully executed.